unity-ggjj/Assets/Scripts/TextDecoder/ActionDecoder.cs

```diff
@@ -627,13 +627,34 @@ private void SET_ACTOR_POSITION(int oneBasedSlotIndex, ActorAssetName actorName)
     private void MODE(GameMode mode)
     {
         DialogueController.GameMode = mode;
-        if (mode == GameMode.CrossExamination)
+        switch (mode)
         {
-            StoryController.OnCrossExaminationStart();
+            case GameMode.Dialogue:
+                PenaltyManager.OnCrossExaminationEnd();
+                break;
+            case GameMode.CrossExamination:
+                PenaltyManager.OnCrossExaminationStart();
+                break;
+            default:
+                Debug.LogError($"Invalid GameMode: {mode}");
+                break;
```

Comment:
Nit: Having this error is actually a good idea! I think this line could actually even be changed to `throw` an exception. I'd With every current game mode now having specific logic executed here, if we're ever in a scenario, where `mode` is set to a value that we don't expect, something has gone wrong bad enough to stopping execution over. 
```suggestion
                throw new NotSupportedException($"Switching to game mode '{mode}' is not supported");
```

(This would also force us to revisit this `switch`, every time we add a new game mode. That way, rather than this method just silently succeeding, we always have to make a conscious decision on whether we want something to happen here for that new game mode. Having something like `case GameMode.NewGameMode: break;` is totally fine, **if** it has been added deliberately.)